rules/library.bzl

```diff
@@ -450,10 +524,58 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
             tags = _MANUAL,
         )
         vendored_deps.append(import_name)
+        import_headers = native.glob(
+            ["**/*.h"],
+            exclude = ["**/.DS_Store"],
+        )
+        import_module_maps = native.glob(
+            ["**/*.modulemap"],
+        )
+        if len(import_module_maps) > 0:
+            import_module_map = import_module_maps[0]
+        else:
+            import_module_map = None
+        vfs_imported_framework = _find_imported_framework_name(import_headers)
+        vfs_framework_name = vfs_imported_framework if vfs_imported_framework else namespace
+        framework_vfs_overlay(
+            name = import_name + "_vfs",
+            framework_name = vfs_framework_name,
+            modulemap = import_module_map,
+            hdrs = import_headers,
+            tags = _MANUAL,
+        )
+        import_vfsoverlays.append(import_name + "_vfs")
+
     for vendored_dynamic_library in kwargs.pop("vendored_dynamic_libraries", []):
         fail("no import for %s" % vendored_dynamic_library)
+        import_headers += native.glob(
+            ["**/*.h"],
+            exclude = ["**/.DS_Store"],
+        )
+        import_module_map = native.glob(
+            ["**/*.modulemap"],
+        )
+
+        if len(import_module_maps) > 0:
+            import_module_map = import_module_maps[0]
+        else:
+            import_module_map = None
+
+        vfs_imported_framework = _find_imported_framework_name(import_headers)
+        vfs_framework_name = vfs_imported_framework if vfs_imported_framework else namespace
+        framework_vfs_overlay(
+            name = import_name + "_vfs",
+            framework_name = vfs_framework_name,
+            modulemap = import_module_map,
+            hdrs = import_headers,
+            tags = _MANUAL,
+        )
+        import_vfsoverlays.append(import_name + "_vfs")
+
     for xcframework in kwargs.pop("vendored_xcframeworks", []):
         vendored_deps.append(_xcframework(library_name = name, **xcframework))
+        # TODO: for virtual frameworks we'll need to add this
```

Comment:
I wonder if we can group pair on this to finish this TODO to see if we truly understand everything haha

Reply:
This is one piece that can be tackled as a followup to keep the size smaller likely

Reply:
@gyfelton Sent an invite offline - I've got a ton of stuff going on this week so it'd be real helpful if you want to jump on this 🤝